Fix Video Unavailable problem at YouTube preview

Sn-Kinos opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->
Fixed the problem that YouTube preview shows Video Unavailable instead of the video.
- Added the properties on the iframe for YouTube preview at `youtube_video.tsx`

#### Reproduce
Create a post that occurs Video Unavailable problem. I used https://youtu.be/edsx_MOhVnk on the below screenshots.

#### Ticket Link
<!--
If applicable, please include both or either of the following links:

Fixes https://github.com/mattermost/mattermost/issues/XXX
Jira https://mattermost.atlassian.net/browse/MM-XXX
-->
No Tickets. (less than 20 lines of code change)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->
I haven't change the UI. But I can show about it w/ screenshots.

|  before  |  after  |
|----|----|
| <img width="459" alt="image" src="https://github.com/mattermost/mattermost/assets/26380261/dde45269-4ad1-450f-9387-1179c12301ca"> | <img width="456" alt="image" src="https://github.com/mattermost/mattermost/assets/26380261/ed9d1cab-eb8c-4cb4-8dec-3e6a63e07b7b"> |

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
Fixed the problem that YouTube preview shows Video Unavailable instead of the video.
```

mattermost-build commented:
Hello @Sn-Kinos,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

pvev commented:
hey @Sn-Kinos , thank you for your contribution. Could you please describe the exact steps on how to reproduce the issue you are trying to solve so we can create a ticket and associate it with this fix? Thanks!

Sn-Kinos commented:
> hey @Sn-Kinos , thank you for your contribution. Could you please describe the exact steps on how to reproduce the issue you are trying to solve so we can create a ticket and associate it with this fix? Thanks!

Thanks @pvev! I added the reproduce section on the text :D

Sn-Kinos commented:
@pvev I wonder if there's anything else I need to do 🏃‍♂️

pvev commented:
> @pvev I wonder if there's anything else I need to do 🏃‍♂️

Thanks @Sn-Kinos we will start the dev review and security review and QE process now. We will keep you posted.

hmhealey commented on webapp/channels/src/components/youtube_video/youtube_video.tsx:
```diff
@@ -75,6 +76,11 @@ export default class YoutubeVideo extends React.PureComponent<Props, State> {
                     height='360px'
                     frameBorder='0'
                     allowFullScreen={true}
+                    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
```
Do we need to allow all these things? `picture-in-picture` and `encrypted-media` are the only ones that seem like they would be applicable to a video in MM. `autoplay` is technically applicable, but I don't think we want autoplaying video in MM

Sn-Kinos replied:
It's necessary and the feature has been on `src` as `autoplay=1`.
| Without autoplay | With autoplay |
| --- | --- |
| <video src="https://github.com/mattermost/mattermost/assets/26380261/f7d1ea07-7a62-47d9-867d-23851b2232d1"> | <video src="https://github.com/mattermost/mattermost/assets/26380261/9d3428b8-eee8-4ad3-99bb-a32e51dded73"> |

hmhealey replied:
Oh right. I forgot we went from showing a static image to the actual video. That makes sense.

I guess web-share/clipboard-write are needed for a share button and accelerometer/gyroscope are needed for rotating the video then

hmhealey reviewed (dismissed):
Thanks for the PR, @Sn-Kinos! I always assumed that the fact that YouTube videos didn't play for me in MM was something on my end, not that we were missing some things on the `iframe`. Great find!

pvev commented:
@Sn-Kinos sorry that this review is taking longer that normal, but there are some security concerns about this change that are being discussed internally. As soon as we define an action, we will let you know. Thanks.

Sn-Kinos commented:
> @Sn-Kinos sorry that this review is taking longer that normal, but there are some security concerns about this change that are being discussed internally. As soon as we define an action, we will let you know. Thanks.

I understand. I referred this iframe code from another messenger app. I hope it can help you.

pvev commented:
@Sn-Kinos, thank you for your patience. The decision to use the referrer policy strict-origin-when-cross-origin can expose some sensitive information, which may be considered more or less critical depending on the use case. Therefore, this decision should be made by the workspace administrator.

Given this, the idea is to make this value configurable through a parameter in the admin console. I wanted to ask if you would like to add this new configuration parameter and set the referrer policy based on that parameter (I would help and guide you through this process). Alternatively, I can merge your PR as it is and then take care of adding this configuration parameter in a subsequent PR.

Sn-Kinos commented:
@pvev I prefer a subsequents PR. I thought there would be no problem because I brought the parameters used in Slack as it is, but I guess it wasn't?

pvev reviewed (approved):
Thank you for the contribution @Sn-Kinos 🚀

pvev commented:
/update-branch

Sn-Kinos commented:
@pvev I fixed lint and type error on CI. How can I re-run actions?

Sn-Kinos commented:
@pvev I fixed code to pass all tests. Is there anything to do for me to merge this?

yasserfaraazkhan reviewed (approved):
before 
![Screenshot 2024-06-11 234834](https://github.com/mattermost/mattermost/assets/15979783/cb461027-6ae6-4c86-b123-22b975fad1a1)

PR changes are working as expected.

![Screenshot 2024-06-11 234851](https://github.com/mattermost/mattermost/assets/15979783/24a21985-23fe-455c-8ee8-8ccd152f9d50)

Sn-Kinos commented:
Ohhhh I found the case that send shorts link. I'll make it now.

Sn-Kinos commented:
Come to think of it, it looks good to make a new PR about YouTube Shorts layout. Should I revert the commit?

hmhealey commented:
> Come to think of it, it looks good to make a new PR about YouTube Shorts layout. Should I revert the commit?

Yeah, if you could revert it and submit it separately, that'd be good. The rest of your changes have gone through reviews and should be all ready to merge, so let's get those merged and add better support for shorts in a separate PR

Sn-Kinos commented:
> > Come to think of it, it looks good to make a new PR about YouTube Shorts layout. Should I revert the commit?
> 
> Yeah, if you could revert it and submit it separately, that'd be good. The rest of your changes have gone through reviews and should be all ready to merge, so let's get those merged and add better support for shorts in a separate PR

I reverted it! Should I made the issue to make new PR for shorts?

hmhealey commented:
> I reverted it! Should I made the issue to make new PR for shorts?

Yeah, please do

Sn-Kinos commented:
I reverted code and passed all tests. Is there anything to do for me to merge this?

larkox commented:
@esarafianou Friendly reminder on this.

larkox commented:
@Sn-Kinos Are you up for adding this change? It will involve some (minor) go code changes (to add the config value on the server side). The idea is to add this new setting, some code in the system console (we should decide where exactly, but we are up for suggestions), and only pass these new props if the config allows them. Does that make sense?

If you are not able to do this, we will figure it out (either a new help wanted ticket, or someone does it internally) but may take some time to get this merged.

Sn-Kinos commented:
@larkox Maybe I can do this. If there are reference, It will be better.

I think this is a bug fix and I was wondering why there should be a configuration for whether or not to provide this feature. Since it is a matter of solving the problem that the original feature is not working, I think that feature should be provided to the user immediately without configuration even if it is delayed a little.

pvev commented:
> @larkox Maybe I can do this. If there are reference, It will be better.
> 
> I think this is a bug fix and I was wondering why there should be a configuration for whether or not to provide this feature. Since it is a matter of solving the problem that the original feature is not working, I think that feature should be provided to the user immediately without configuration even if it is delayed a little.

@Sn-Kinos thank you for solving this issue, but sharing the referral could be consider sharing sensitive information. That's why we have decided that it should be responsibility of the workspace admin to decide if they want to share the referrer information. We want to set the expose of the referrer behind a config value under `Experimental > Features` as '`YouTube Referrer Policy`'. This would be a boolean value, with the default set to false, meaning the referrer policy remains `no-referrer`, but when set to true the referrer policy would change to `strict-origin-when-cross-origin`. An example of creating the admin config can be found here https://github.com/mattermost/mattermost/pull/22952 . Let me know if you have questions around it.

Sn-Kinos commented:
> > @larkox Maybe I can do this. If there are reference, It will be better.
> > I think this is a bug fix and I was wondering why there should be a configuration for whether or not to provide this feature. Since it is a matter of solving the problem that the original feature is not working, I think that feature should be provided to the user immediately without configuration even if it is delayed a little.
> 
> @Sn-Kinos thank you for solving this issue, but sharing the referral could be consider sharing sensitive information. That's why we have decided that it should be responsibility of the workspace admin to decide if they want to share the referrer information. We want to set the expose of the referrer behind a config value under `Experimental > Features` as '`YouTube Referrer Policy`'. This would be a boolean value, with the default set to false, meaning the referrer policy remains `no-referrer`, but when set to true the referrer policy would change to `strict-origin-when-cross-origin`. An example of creating the admin config can be found here #22952 . Let me know if you have questions around it.

Thanks! I understand about it. I can work for it. How do I write the description?

pvev commented:
> > > @larkox Maybe I can do this. If there are reference, It will be better.
> > > I think this is a bug fix and I was wondering why there should be a configuration for whether or not to provide this feature. Since it is a matter of solving the problem that the original feature is not working, I think that feature should be provided to the user immediately without configuration even if it is delayed a little.
> > 
> > 
> > @Sn-Kinos thank you for solving this issue, but sharing the referral could be consider sharing sensitive information. That's why we have decided that it should be responsibility of the workspace admin to decide if they want to share the referrer information. We want to set the expose of the referrer behind a config value under `Experimental > Features` as '`YouTube Referrer Policy`'. This would be a boolean value, with the default set to false, meaning the referrer policy remains `no-referrer`, but when set to true the referrer policy would change to `strict-origin-when-cross-origin`. An example of creating the admin config can be found here #22952 . Let me know if you have questions around it.
> 
> Thanks! I understand about it. I can work for it. How should I write the description?

The description for the system console config value should be something like:

> _Share referrer information when embedding YouTube videos_


And the PR descriptions could be then something like:
_

> This PR introduces a new config for the YouTube referrer policy. The feature allow admins to control whether to expose the referrer information when embedding YouTube videos. By default, the policy is set to false (value is no-referrer) so that no sensitive information is shared. Admins can then change the setting to true (sharing the referrer meaning the value is: strict-origin-when-cross-origin) if they wish to include the referrer information that will unblock the visualization of some youtube videos. The configuration is located under Experimental > Features as 'YouTube Referrer Policy'.

_

cc: @hmhealey

larkox commented on webapp/channels/src/components/admin_console/admin_definition.tsx:
```diff
@@ -6258,6 +6258,13 @@ const AdminDefinition: AdminDefinitionType = {
+                            type: 'bool',
+                            key: 'ExperimentalSettings.YoutubeReferrerPolicy',
+                            label: defineMessage({id: 'admin.experimental.youtubeReferrerPolicy.title', defaultMessage: 'YouTube Referrer Policy:'}),
+                            help_text: defineMessage({id: 'admin.experimental.youtubeReferrerPolicy.desc', defaultMessage: 'When true, the referrer policy for embedded YouTube videos will be set to "strict-origin-when-cross-origin". When false, the referrer policy will be set to "no-referrer".'}),
```
@cwarnermm Can you check these lines?

I guess we should add the final objective of this, which is that this may solve issues with youtube videos previews, but may have some privacy implications.

You may be able to sync with @esarafianou privacy implications.

cwarnermm replied:
@larkox - I propose the following System Console help text update, subject to @esarafianou's input and approval:

"When true, the referrer policy for embedded YouTube videos will be set to "strict-origin-when-cross-origin" which balances the need to protect user privacy with some degree of referral data to support web functionalities, like analytics, logging, and third-party integrations. When false, the referrer policy will be set to "no-referrer" which enhances user privacy by not disclosing the source URL, but limits the ability to track user engagement and traffic sources in analytics tools."

Sn-Kinos replied:
> @pvev I prefer a subsequents PR. I thought there would be no problem because I brought the parameters used in Slack as it is, but I guess it wasn't?

I just use parameters from Slack. If there are useless parameter and reply it, I'll remove it.

Sn-Kinos replied:
I think it should contains the content that it can solves the problem that YouTube video preview shows `Video Unavailable`.

cwarnermm replied:
@Sn-Kinos - which referrer policy results in the ``Video Unavailable`` being resolved?

larkox replied:
@cwarnermm `no-referrer` (what we used to have) can result in Youtube marking the videos as unavailable. The `strict-origin-when-cross-origin` one solves the issue, but as you said, it exposes information.

cwarnermm replied:
@Sn-Kinos - I've updated my [proposed text update](https://github.com/mattermost/mattermost/pull/26980#discussion_r1671176960) above to incorporate your feedback.

larkox commented on webapp/channels/src/components/youtube_video/youtube_video.test.tsx:
```diff
@@ -31,6 +31,7 @@ describe('YoutubeVideo', () => {
                 url: 'linkForThumbnail',
             }],
         },
+        youtubeReferrerPolicy: false,
```
Can we add one test (even if it is a snapshot test) with this setting set to true?

larkox commented on webapp/channels/src/components/youtube_video/youtube_video.tsx:
```diff
@@ -75,6 +77,10 @@ export default class YoutubeVideo extends React.PureComponent<Props, State> {
                     height='360px'
                     frameBorder='0'
                     allowFullScreen={true}
+                    allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share'
+                    referrerPolicy={this.props.youtubeReferrerPolicy ? 'strict-origin-when-cross-origin' : undefined}
+                    title={videoTitle}
+                    sandbox='allow-scripts allow-same-origin allow-popups allow-presentation'
```
Apart of the referrerPolicy... The rest of these props are needed? Just for my own knowledge, do you mind explaining what they do and how it differ from the default value?

@esarafianou Just in case you have any concern about any of these other props.

Sn-Kinos replied:
[I just uses parameters from Slack.](https://github.com/mattermost/mattermost/pull/26980#issuecomment-2155201354) If there are useless parameter and reply it, I'll remove it.

larkox commented on webapp/channels/webpack.config.js:
```diff
@@ -411,6 +411,8 @@ if (targetIsDevServer) {
                 '/api': {
                     ...proxyToServer,
                     ws: true,
+                    changeOrigin: true,
+                    secure: false,
```
Just for my own knowledge, what are these changes for?

Sn-Kinos replied:
Oh sorry, It's my mistake. That is the config for my local dev env.

larkox reviewed (commented):
Looks great. Nothing really blocking, just a few questions, mostly for my knowledge.

pvev commented on e2e-tests/playwright/support/server/default_config.ts:
```diff
@@ -16,7 +16,7 @@ import {
 } from '@mattermost/types/config';
 import testConfig from '@e2e-test.config';
-
+    
```
unnecessary line

pvev commented on webapp/channels/src/components/youtube_video/youtube_video.tsx:
```diff
@@ -75,6 +77,10 @@ export default class YoutubeVideo extends React.PureComponent<Props, State> {
                     frameBorder='0'
                     allowFullScreen={true}
+                    allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share'
+                    referrerPolicy={this.props.youtubeReferrerPolicy ? 'strict-origin-when-cross-origin' : undefined}
```
shouldn't we better set this directly to 'no-referrer' ?

Sn-Kinos replied:
I chose to use it indirectly to avoid having to find and change these parts one by one when the `Referrer-Policy` set globally changes. If 'Referrer-Policy' does not change no matter what, I will change it from `undefined` to `no-referrer`

pvev replied:
it shouldn't change. Even for clarity starting it with `no-referrer` works better for me.

Sn-Kinos replied:
I understood that YouTube should remain 'no-ferrer' even if the global settings change. I'll change it.

pvev reviewed (approved):
thank you very much for all the effort on this PR @Sn-Kinos 🚀 🎉 , it looks good to me now. Just two small comments.

yasserfaraazkhan commented:
/update-branch

yasserfaraazkhan commented:
Waiting for CI failures to be fixed.
Will start QA after that.

hmhealey commented on webapp/channels/src/i18n/ko.json:
```diff
@@ -768,6 +768,8 @@
   "admin.experimental.userStatusAwayTimeout.desc": "이 설정은 사용자가 Mattermost 사용 중 자리를 비울 때 사용자 상태 표시를 \"자리 비움\"으로 변경되는 시간(초)을 정의합니다.",
   "admin.experimental.userStatusAwayTimeout.example": "예: \"300\"",
   "admin.experimental.userStatusAwayTimeout.title": "사용자 상태 자리 비움 시간초과 :",
+  "admin.experimental.youtubeReferrerPolicy.desc": "활성화 시, 임베디드 유튜브 비디오의 Referrer 정책이 \"strict-origin-when-cross-origin\"로 설정되어 유튜브 비디오 미리보기가 사용할 수 없음으로 표시되는 문제를 해결하며 사용자의 개인 정보 보호를 보호하면서 웹 기능(분석, 로깅 및 타사 통합)을 지원하기 위한 일부 참조 데이터를 제공합니다. 비활성화 시, Referrer 정책이 \"no-referrer\"로 설정되어 사용자의 개인 정보 보호를 강화하고 소스 URL을 공개하지 않지만 분석 도구에서 사용자 참여 및 트래픽 소스를 추적하는 능력이 제한됩니다.",
+  "admin.experimental.youtubeReferrerPolicy.title": "유튜브 Referrer Policy 설정 :",
```
@Sn-Kinos Would you mind removing these? We appreciate the help translating these, but the translation tool we use runs into conflicts if someone changes the non-English translation files directly

Sn-Kinos commented:
> Waiting for CI failures to be fixed. Will start QA after that.

Should I update the snapshot of `youtube_video.test.tsx.snap`? I didn't changes it but its snapshot was changed.

![image](https://github.com/user-attachments/assets/02d22093-5a89-420a-aa83-d9cbe09a9752)
![image](https://github.com/user-attachments/assets/c5f2bb19-3a2f-4a76-abb5-7be6cbb35107)

pvev commented:
> > Waiting for CI failures to be fixed. Will start QA after that.
> 
> Should I update the snapshot of `youtube_video.test.tsx.snap`? I didn't changes it but its snapshot was changed.
> 
> ![image](https://private-user-images.githubusercontent.com/26380261/348960178-02d22093-5a89-420a-aa83-d9cbe09a9752.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMTgzMjMsIm5iZiI6MTcyMTExODAyMywicGF0aCI6Ii8yNjM4MDI2MS8zNDg5NjAxNzgtMDJkMjIwOTMtNWE4OS00MjBhLWFhODMtZDljYmUwOWE5NzUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDA4MjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZjU4MDM1ZDNkNGVjNzQ2ZTc2MTZlOTIzNmJiZTE5ZmUyOGMwY2IzMzJlMmIwNTVjNDBlYTRhNTkxODljY2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gpM_HSjyb6EnKajoM_4Wt6V_hpyLL-FB5ClfpHDrIoM) ![image](https://private-user-images.githubusercontent.com/26380261/348960203-c5f2bb19-3a2f-4a76-abb5-7be6cbb35107.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMTgzMjMsIm5iZiI6MTcyMTExODAyMywicGF0aCI6Ii8yNjM4MDI2MS8zNDg5NjAyMDMtYzVmMmJiMTktM2EyZi00YTc2LWFiYjUtN2JlNmNiYjM1MTA3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDA4MjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2N2QxMTc0MGUzNTVmZTMxN2YyZjAyYzNhNjZhNTQ1YWRjZDdlNDAyODA2OTI1NzBmODBmZTMwZmEzNjM2NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8PRCLxSovD5vkC6Dne9C-IKq27sZSxWwKYRPFLxwnU8)

yes please, you can update the snapshots using the `npm run test -u {path}` command. Thanks.

hmhealey reviewed (approved):
Thanks for all the work on this. This looks great!

hmhealey commented:
Just a heads up, I removed an outdated snapshot for a test you added and then removed. Hopefully that should fix CI

Sn-Kinos commented:
> Just a heads up, I removed an outdated snapshot for a test you added and then removed. Hopefully that should fix CI

I added it according to [this comment](https://github.com/mattermost/mattermost/pull/26980#discussion_r1670009357). Even though I've seen the CLI say it's obsolete, is it okay to remove it?

pvev reviewed (approved):
thank you for this awesome contribution @Sn-Kinos 🚀 , it looks perfect to me now!

hmhealey commented:
> > Just a heads up, I removed an outdated snapshot for a test you added and then removed. Hopefully that should fix CI
> 
> I added it according to [this comment](https://github.com/mattermost/mattermost/pull/26980#discussion_r1670009357). Even though I've seen the CLI say it's obsolete, is it okay to remove it?

Yeah, I think so. It looks like that's an extra test that you added in https://github.com/mattermost/mattermost/pull/26980/commits/fc7886331f63293bb2b5a9307891cfbfcca76287 which wasn't actually needed. That commit adds only one new test, but it includes snapshots for two different tests. I just removed the second one which didn't have an accompanying test

Sn-Kinos commented:
It passed all tests but test server creation failed. Is there anything to do for me to solve this problem?

yasserfaraazkhan commented:
/update-branch

Sn-Kinos commented:
May I access the test server to test?

yasserfaraazkhan commented:
> May I access the test server to test?

@Sn-Kinos Yes you can

yasserfaraazkhan reviewed (changes requested):
when we play the YT video, it says unavailable.

<img width="632" alt="image" src="https://github.com/user-attachments/assets/f0d825e1-e876-4866-a30c-416b69550834">

Sn-Kinos commented:
@yasserfaraazkhan Setting was added by request. I checked it was false and I changed.
<img width="905" alt="image" src="https://github.com/user-attachments/assets/a69c017a-f52f-4ad7-ae31-b47510aa9b1d">

But it didn't work when I changed the settings. I checked and found a case typo and fixed it.

<img width="956" alt="image" src="https://github.com/user-attachments/assets/4f2cfbda-bba3-4875-9600-eea4a2f783e7">

I checked it works with command `MM_SERVICESETTINGS_SITEURL=https://mattermost-pr-26980.test.mattermost.cloud/ npm run dev-server`

Sn-Kinos commented:
I just fixed typo but CI fails a lot... 😨 What can I do for solve it?

larkox commented:
/update-branch

larkox commented:
@Sn-Kinos Seems like a hiccup in the CI. Let's see if re-running the CI solves it

yasserfaraazkhan commented:
/update-branch

yasserfaraazkhan reviewed (approved):
able to play the video now.
Thank you @Sn-Kinos

Sn-Kinos commented:
I'm waiting to develop YouTube Shorts layout. Is it only the Security Review left now?

larkox commented:
@esarafianou Can you confirm this change?

esarafianou reviewed (approved):
@larkox code looks good but we need someone from Product to agree on adding a new experimental option since we now better track the different product lifecycle paths (experimental - alpha - beta - GA). Adding it as under `ExperimentalSettings` is another thing we need Product to agree on as this makes moving the option out of experimental a breaking change (needing to change the config name).

@wiersgallak  can you let us know what you think?

wiersgallak commented:
@esarafianou From what I understand, as long as the word "experimental" is not included in the configuration name and it is just under the experimental config setting location, we can move the config setting location without changing the full name of it.

Sn-Kinos commented:
> @esarafianou From what I understand, as long as the word "experimental" is not included in the configuration name and it is just under the experimental config setting location, we can move the config setting location without changing the full name of it.

Is there anything to do for me to solve it?

pvev commented:
@Sn-Kinos I think this is ready now to merge. Thank you very much for your contribution and for your patience during our review. Our QE process makes exhaustive reviews to any bit we will put in our code base to always guarantee the best quality and security.
